dagql/README.md

```diff
@@ -0,0 +1,69 @@
+# dagql
+
+DagQL is a strongly opinionated implementation of a GraphQL server.
+
+## axioms
+
+Below are a set of assertions that build on one another.
+
+* All Objects are immutable.
+* All Objects are [Nodes][Node], i.e. all objects have an `id`.
+* All Objects have their own ID type, e.g. `PointID`.
+* All Objects have a top-level constructor named after the object, e.g. `point`.
+* All Objects may be loaded from an ID, which will create the Object if needed.
+* An Object's field may be `@impure` which indicates that the field's result shall not be cached.
+* An Object's field may be `@meta` which indicates that the field may be omitted without affecting the result.
+* All IDs are derived from the query that constructed the Object.
+* An ID is *canonicalized* by removing any embedded `@meta` selectors.
+* An ID is *impure* if it contains any `@impure` selectors or any *tainted* IDs.
+* An ID may be loaded on a server that has never seen its Object before.
```

Comment:
This includes IDs with local dir blobs, in memory secrets, etc.? If so, then I guess while it can be loaded on a server, that doesn't mean it necessarily works right? Just checking

Reply:
Yep true